common/djangoapps/entitlements/models.py

```diff
@@ -1,8 +1,82 @@
 import uuid as uuid_tools
+from datetime import datetime, timedelta
 
+import pytz
 from django.conf import settings
+from django.contrib.sites.models import Site
 from django.db import models
+
 from model_utils.models import TimeStampedModel
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class CourseEntitlementPolicy(models.Model):
+    """
+    Represents the Entitlement's policy for expiration, refunds, and regaining a used certificate
+    """
+
+    expiration_period_days = models.IntegerField(
+        default=450,
+        help_text="Number of days from when an entitlement is created until when it is expired.",
+        null=False
+    )
+    refund_period_days = models.IntegerField(
+        default=60,
+        help_text="Number of days from when an entitlement is created until when it is no longer refundable",
+        null=False
+    )
+    regain_period_days = models.IntegerField(
+        default=14,
+        help_text=("Number of days from when an entitlement is redeemed for a course run until "
+                   "it is no longer able to be regained by a user."),
+        null=False
+    )
+    site = models.ForeignKey(Site)
+
+    def get_days_until_expiration(self, entitlement):
```

Comment:
Nit: We use the term `entitlement` here when I think we actually mean `course_entitlement`. I think it would be good if we could avoid conflating these terms as they refer to different tables/concepts.

Reply:
I think I'm going to leave this, in the context of the course_entitlement objects I think the class methods just referencing itself is clear enough without creating an overly verbose method call.

Reply:
Should this take into account the other logic necessary for determining expiration, like whether or not the entitlement has been spent? Otherwise, it will probably be inaccurate, as an entitlement that has been spent should not have > 14 days until expiration.